From 82f55d60c32974b990729b9b68a2440ead0ff50c Mon Sep 17 00:00:00 2001
From: Prabhath P Raman <prabhath@broadcom.com>
Date: Fri, 9 Nov 2012 12:29:31 +0530
Subject: [PATCH 420/565] XLP Network driver: Port dynamic loadbalancing
 support

Added support for dynamic flow migration between cpus. With this
feature all TCP flows are distributed uniformly to all available
cpus without modifying the application.

Based on Broadcom SDK 2.3.

Signed-off-by: Prabhath P Raman <prabhath@broadcom.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/Kconfig | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/mips/netlogic/Kconfig b/arch/mips/netlogic/Kconfig
index cf159c7..da9bf45 100644
--- a/arch/mips/netlogic/Kconfig
+++ b/arch/mips/netlogic/Kconfig
@@ -168,3 +168,8 @@ config PCI_XLP_MSI
 	bool
 	depends on NLM_XLP && PCI && PCI_MSI
 	default y
+
+config NLM_ENABLE_LOAD_BALANCING
+	bool "Enable load balancing in nae driver for TCP_PERF mode."
+	depends on NLM_XLP && 64BIT && !NUMA
+	default n
-- 
1.8.4.93.g57e4c17

